modules/face/tutorials/face_landmark_detection.markdown

```diff
@@ -0,0 +1,160 @@
+
+To get more detailed description about the training parameters you can refer to the [Research paper](https://pdfs.semanticscholar.org/d78b/6a5b0dcaa81b1faea5fb0000045a62513567.pdf).
+
+Understanding code
+-------------------
+
+For understanding the code jump to this [page](face_landmarks_trainer/face_landmark_trainer.markdown)
+
+Error rate
+----------
+
+**The error rate on trained images depends on the number of images used for training used as follows :**
+
+![](face_images/train.png)
+
+**The error rate on test images depends on the number of images used for training used as follows :**
+
+![](face_images/test.png)
+
+Face landmark detection in an image
+===================================
+
+![](face_images/facereg.jpg)
+
+This application lets you detect landmarks of detected faces in an image. You can detect landmarks of all the faces found in an image
+and use them further in various applications like face swapping, face averaging etc.
+This functionality is now available in OpenCV.
+
+```
+// Command to be typed for running the sample
+./sampleDetectLandmarks -file=trained_model.dat -face_cascade=lbpcascadefrontalface.xml -image=/path_to_image/image.jpg
+```
+Description of command parameters
+---------------------------------
+
+> * **model_filename** f : (REQUIRED) A path to binary file storing the trained model which is to be loaded [example - /data/file.dat]
+> * **image** i : (REQUIRED) A path to image in which face landmarks have to be detected.[example - /data/image.jpg]
+> * **face_cascade** c : (REQUIRED) A path to the face cascade xml file which you want to use as a face detector.
+
+Understanding code
+------------------
+
+For understanding the code jump to this [page](face_landmark_detection/face_landmark_detection.markdown)
+
+Face landmark detection in a video
+===================================
+
+Sample video:
+
+@htmlonly
+<iframe width="420" height="315" src="https://youtu.be/dZXKsrApRJk" frameborder="0" allowfullscreen></iframe>
+@endhtmlonly
+
+This application lets you detect landmarks of detected faces in a video.This application first detects faces in a current video frame
+and then finds their facial landmarks. You just have to pass the video as input.
+```
+// Command to be typed for running the sample
+./sampleDetectLandmarks -file=trained_model.dat -face_cascade=lbpcascadefrontalface.xml -video=/path_to_video/video.avi
+```
+Description of command parameters
+---------------------------------
+
+> * **model_filename** f : (REQUIRED) A path to binary file storing the trained model which is to be loaded [example - /data/file.dat]
+> * **video** v : (REQUIRED) A path to video in which face landmarks have to be detected.[example - /data/video.avi]
+> * **face_cascade** c : (REQUIRED) A path to the face cascade xml file which you want to use as a face detector.
+
+Understanding code
+------------------
+
+For understanding the code jump to this [page](face_landmark_detection_video/face_landmark_video.markdown)
```

Comment:
@mshabunin @alalek does any one of your know if OpenCV doxygen allows for multi page tutorials? After looking at the warnings and the pre-generated test view of the docs, it seems this one is missing (and thus I guess the warning originate from here). The only direct difference I see is the explicit page linking.

Furthermore 2 questions
1. How do we enable the fact that FaceMark class is visible in face module overview in docs. For now it is not showing there.
2. Do we need to explicitly link the tutorials somewhere? Or is providing an overview page more than enough inside the tutorials folder?

Reply:
@StevenPuttemans , cmake gathers all top-level pages in `<module>/tutorials` folders and lists them on the "Tutorials for contrib modules" page.
Module author can either put all tutorials into subfolder and create single top-level index page (_bioinspired_ module), either leave all tutorials on the top-level (_tracking_ module). We do not have any guidelines about it yet.
As I can see, only the face_landmark_trainer.markdown_ tutorial should be added to the index page.

Reply:
For now we solved it by providing a face link at top level. That page gives an overview with tutorials for both the recognition and the landmark detection part. This was the cleanest instead of having multiple face entries at the top level page.